x-pack/filebeat/input/http_endpoint/config.go

```diff
@@ -79,9 +87,29 @@ func (c *config) Validate() error {
 		return errors.New("hmac.type must be sha1 or sha256")
 	}
 
+	if c.CRCProvider != "" {
+		err := validateCRCProvider(c.CRCProvider)
+		if err != nil {
+			return err
+		} else if c.SecretValue == "" {
+			return errors.New("secret.value is required when crc.provider is defined")
+		}
+	}
+
 	return nil
 }
 
+func validateCRCProvider(value string) error {
+	value = strings.ToLower(value)
+	for _, v := range validCRCProviders {
+		if strings.ToLower(v) == value {
```

Comment:
Not for here since the map approach would be preferred, but when doing case-insensitive string comparison, the preferred approach is to use `strings.EqualFold`.